src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/TimeSpanConverter.cs

```diff
@@ -54,7 +54,7 @@ public override TimeSpan Read(ref Utf8JsonReader reader, Type typeToConvert, Jso
                 int backslash = source.IndexOf(JsonConstants.BackSlash);
                 Debug.Assert(backslash != -1);
 
-                Span<byte> sourceUnescaped = stackalloc byte[source.Length];
+                Span<byte> sourceUnescaped = stackalloc byte[MaximumEscapedTimeSpanFormatLength];
```

Comment:
Can't comment on the exact location, but shouldn't line 33 on this file also be changed to 
```csharp
Span<byte> stackSpan = stackalloc byte[MaximumEscapedTimeSpanFormatLength];
```

Reply:
Shouldn't this be
```suggestion
                Span<byte> sourceUnescaped = stackalloc byte[MaximumTimeSpanFormatLength];
```
given that it's writing the unescaped form?

Reply:
* Line 38: This originally came from https://github.com/dotnet/runtime/pull/54186#discussion_r662533772. Happy to change if you want. Not sure what will have best perf, checking the flag or allocating MaximumEscapedTimeSpanFormatLength (x3) when we know MaximumTimeSpanFormatLength is all we need?

* Line 57: I made the change but the tests blow up. There's this assert in Unescape:

https://github.com/dotnet/runtime/blob/188c2702cd6f365bfad3a985f82e1f9cd520aeb4/src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.Unescaping.cs#L327

Reply:
- Line 38: we can leave it be for now and make a call later.
- Line 57: I might have misinterpreted how that particular buffer is being used. Sorry about that.